### DISCUSSION TITLE
Singular EXE build

### DISCUSSION THREAD

**Discussion by Zariffs** [tauri-apps/tauri#11167] [Q&A]
is there some kind of way to build tauri into just one exe file with the assets folder and a lib folder? lib for things like dlls and webView2 Files and settings? is this possible? would really appriciate some help

**Comment by FabianLars** [comment]
If you need to bundle files like dlls (which can't really be stored inside the exe) then no.
For "normal" Tauri apps this is generally possible but has a few (potentionally major) caveats:
- You will lose access to a few features: deep links, file associations, sidecar, resources, updater
  - Those are the ones i can think of right now at least
  - New features in the future may also require bundles/installers
- The notifications may not be displayed, or more likely, be displayed but with a PowerShell icon
- In my opinion the most important one: The installer is responsible for making sure that webview2 is installed on the system.
  - Only Windows 11 had it since release so that's the only system you can expect to work oob (ignoring edgecases where power users managed to uninstall it)
  - Windows 10 also got it via an update but relying on users keeping their system up to date is just asking for trouble lol
  - You can install it on first app start in rust before the `tauri::Builder` line but that's imo not a nice user experience
    - example: https://discord.com/channels/616186924390023171/618328921330417692/1028411797323120711 (idk if it's still 100% correct after 2 years)

To get the .exe, simply build your tauri app as usual. The .exe will be in `target/release/` (_not_ in the `bundle` subfolder).

P.S. i assume we're talking about windows only from your mention of webview2 and the fact that single exe/binary apps aren't really a thing on macos and linux
